doc/user-guide.md

```diff
@@ -11,189 +59,301 @@ funcool/rumext
+(let [props #js {:className "fooBar"
+                 :style #js {:backgroundColor "red"}
+                 :onClick some-on-click}]
+  [:> "div" props
+    "Hello World"])
 ```
 
-**NOTE**: Important: the `*` in the name is a mandatory convention for
-proper visual distinction of React components and Clojure functions.
+**IMPORTANT** if you define the attributes dynamically, outside the `:>` macro,
+there are no automatic transformations. So you need to define the map as a
+plain Javascript object with the `#js` prefix or any other way. You also need
+to use `camelCase` names and remember to use `className` instead of `class`,
+for example.
 
-**NOTE**: It also enables the current defaults on how props are
-handled. If you don't use the `*` suffix, the component will behave in
-legacy mode.
+There are a couple of utilities for managing dynamic attributes in a more
+convenient way.
 
-## Props & Destructuring
+#### `mf/spread`
 
-The destructuring works very similar to the Clojure map destructuring
-with small differences and convenient enhancements for making working
-with React props and idioms easy.
+A macro that allows performing a merge between two props data structures using
+the JS spread operator (`{...props1, ...props2}`). This macro also performs
+name transformations if you pass a literal map as a second parameter.
 
-```clojure
-(mf/defc title*
-  [{:keys [name] :as props}]
-  (assert (object? props) "expected object")
-  (assert (string? name) "expected string")
-  (assert (= (unchecked-get props "name")
-             name)
-          "expected string")
+It is commonly used this way:
 
-  [:label {:class "label"} name])
+```clojure
+(mf/defc my-label*
+  [{:keys [name class on-click] :rest props}]
+  (let [class (or class "my-label")
+        props (mf/spread props {:class class})]
+    [:span {:on-click on-click}
+      [:> :label props name]]))
 ```
 
-An additional idiom (specific to the Rumext component macro and not
-available in standard Clojure destructuring) is the ability to obtain
-an object with all non-destructured props:
+#### `mf/props`
+
+A helper macro to create a Javascript props object from a Clojure map,
+applying name transformations.
+
+An example of how it can be used and combined with `mf/spread`:
 
 ```clojure
-(mf/defc title*
-  [{:keys [name] :rest props}]
-  (assert (object? props) "expected object")
-  (assert (nil? (unchecked-get props "name")) "no name in props")
+(mf/defc my-label*
+  [{:keys [name class on-click] :rest props}]
+  (let [class (or class "my-label")
+        new-props (mf/props {:class class})
+        all-props (mf/spread props new-props)]
+    [:span {:on-click on-click}
+      [:> :label props name]]))
+```
+
+The map, being a Clojure object, may also be created elsewhere and passed
+to `mf/props` as a variable:
 
-  ;; The `:>` will be explained later
+```clojure
+(let [clj-props {:class "my-label"}
+      props (mf/props clj-props)]
```

Comment:
This code is invalid

Reply:
On the need of dynamic props transformation, `rumext.v2.util/map->props` can be used, but not recommended for performance reasons of performing that transformation at runtime

Reply:
This example was in the earlier version of the document. Has this changed?